### ISSUE TITLE
+=: on ArrayBuffer can be a no-op

### ISSUE THREAD

**Issue by Ichoran** [scala/collection-strawman#400]
ArrayBuffer doesn't prepend an element when the `+=:` method is called.  It instead slides everything down but doesn't increase the length (as far as I can tell)

```
val b = ArrayBuffer.empty[Int]
1 +=: b
b
```

demonstrates the problem.

```
val b = ArrayBuffer(2)
1 +=: b
b
```

gives `ArrayBuffer(1)`.  Probably the length isn't being updated?

**Comment by Ichoran** [comment]
Issue #411 might be the deeper problem that is causing this one.

**Comment by julienrf** [comment]
Fixed by #417.
